src/LuaTransformer.ts

```diff
@@ -4167,6 +4229,56 @@ export class LuaTransformer {
         }
     }
 
+    public transformLuaTableCallExpression<T extends boolean>(
+        expression: ts.CallExpression,
+        isWithinExpressionStatement: T
```

Comment:
I don't like this parameter. A transformation of an expression should always return an expression. If you need a statement that is something I think you should handle at the call site.